cdc/sink/codec/canal_flat.go

```diff
@@ -30,6 +32,8 @@ type CanalFlatEventBatchEncoder struct {
 	builder       *canalEntryBuilder
 	unresolvedBuf []*canalFlatMessage
 	resolvedBuf   []*canalFlatMessage
+	// When it is true, checkpoint events are generated.
+	waterMark bool
```

Comment:
```suggestion
	watermark bool
```

Just _watermark_, the term is widely used.